src/spi.rs

```diff
@@ -266,29 +350,40 @@ macro_rules! spi {
                         w.mbr()
                             .variant(mbr) // master baud rate
                             .dsize()
-                            .bits(8 - 1) // 8 bit frames
+                            .bits(config.frame_size - 1)
                     });
 
                     // ssi: select slave = master mode
                     spi.cr1.write(|w| w.ssi().slave_not_selected());
 
+                    // Calculate the CS->transaction cycle delay bits.
+                    let mut cycle_delay = (config.cs_delay * spi_freq as f32) as u32;
+
+                    // The calculated cycle delay may not be more than 4 bits wide for the
+                    // configuration register.
+                    if cycle_delay > 0xF {
```

Comment:
I'm not a massive fan of this check, but I don't see a way to get at the maximum value in the MSSI register from the `svd2rust` PAC.

Reply:
Hardware specific constants are completely fine here. We could even panic if the cycle delay isn't possible